src/ksm/client/distributed_pod_label_discovery.go

```diff
@@ -6,27 +6,26 @@ import (
 	"time"
 
 	"github.com/newrelic/infra-integrations-sdk/log"
-	"github.com/pkg/errors"
 	v1 "k8s.io/api/core/v1"
 
 	"github.com/newrelic/nri-kubernetes/v2/src/client"
 )
 
 type distributedPodLabelDiscoverer struct {
-	ksmPodLabel string
-	ownNodeIP   string
-	logger      log.Logger
-	k8sClient   client.Kubernetes
+	ksmPodLabel  string
+	ksmNamespace string
+	ownNodeIP    string
+	logger       log.Logger
+	k8sClient    client.Kubernetes
 }
 
 func (p *distributedPodLabelDiscoverer) findAllLabeledPodsRunningOnNode() ([]v1.Pod, error) {
-	pods, err := p.k8sClient.FindPodsByLabel(p.ksmPodLabel, "true")
-
+	pods, err := p.k8sClient.FindPodsByLabel(p.ksmNamespace, fmt.Sprintf("%s=true", p.ksmPodLabel))
 	if err != nil {
-		return nil, errors.Wrap(err, "could not query api server for pods")
+		return nil, fmt.Errorf("querying API server for pods: %w", err)
 	}
 	if len(pods.Items) == 0 {
-		return nil, errors.Wrapf(errNoKSMPodsFound, "no KSM pod found with label: '%s'", p.ksmPodLabel)
+		return nil, fmt.Errorf("no KSM pod found with label %q in namespace %q: %w", p.ksmPodLabel, p.ksmNamespace, errNoKSMPodsFound)
```

Comment:
Wrapping will make the error message look like:

`no KSM pod found with label %q in namespace %q: no KSM pods found`

Which seems redundant.

What about ...?

```suggestion
		return nil, fmt.Errorf("discovering KSM with label %q in namespace %q: %w", p.ksmPodLabel, p.ksmNamespace, errNoKSMPodsFound)
```

Reply:
Sure.